test/controllers/graphql_controller_test.rb

```diff
@@ -15,4 +15,97 @@ class GraphqlControllerTest < ActionDispatch::IntegrationTest
     post graphql_url, params: { variables: variables, query: query }
     assert_response :success
   end
+
+  context 'authorized user' do
+    setup do
+      account = FactoryBot.create(:account)
+      @current_user = FactoryBot.create(:user, account: account)
+      stub_rbac_permissions(Rbac::COMPLIANCE_VIEWER, Rbac::INVENTORY_VIEWER)
+    end
+
+    should 'be allowed to read inventory' do
+      assert_equal @current_user.authorized_to?(Rbac::INVENTORY_VIEWER), true
+    end
+  end
+
+  context 'unauthorized user' do
+    setup do
+      account = FactoryBot.create(:account)
+      @current_user = FactoryBot.create(:user, account: account)
+      stub_rbac_permissions
+    end
+
+    should 'not be allowed to read inventory' do
+      account = FactoryBot.create(:account)
+      user = FactoryBot.create(:user, account: account)
+      identity_header = IdentityHeader.new(account.b64_identity)
+      GraphqlController.any_instance.stubs(:identity_header).returns(identity_header)
+      IdentityHeader.any_instance.stubs(:valid?).returns(true)
```

Comment:
Instead of `any_instance` mocking, you can simply [pass](https://twobitlabs.com/2010/09/setting-request-headers-in-rails-functional-tests/) the right identity header as a HTTP header below.